pkg/instrumentation/router.go

```diff
@@ -25,16 +25,21 @@ type Tracer struct {
 	Enabled     bool
 	Environment string
 	Options     []tracer.StartOption
+
+	globalServiceName string
```

Comment:
[q]: could be named just `serviceName`? `global` prefix looks a bit redundant here, wdyt?

Reply:
Good observation. I had 2 reasons for that. So in current [library](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.40.1/ddtrace/tracer) there is no shareable object as tracer and program treats them all as global, which is a bad design, And I believe the name should include `global` to give the reader an idea how to deal with such odd case.
second reason, datadog engineers found out it's a bad design and are working on a not ready yet replacement for tracer called [opentracing](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.40.1/ddtrace/opentracer#New) and they address the issue there; technically, they will deprecate tracer and so as we will.
I left this comment open to continue, feel free to resolve it if you're convinced.

Reply:
@laynax Got this, thank you for the clarifications 🙏 

>datadog engineers found out it's a bad design and are working on a not ready yet replacement for tracer called opentracing and they address the issue there

Could you please share the link where this finding was revealed by DataDog engineers? Couldn't find  this in google

Reply:
@Neurostep 
[qoute:](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.40.1/ddtrace/opentracer#New:~:text=Package%20opentracer%20is%20in%20%22Maintenance%22%20mode%20and%20limited%20support%20is%20offered.%20Please%20consider%20using%20OpenTelemetry%20or%20ddtrace/tracer%20directly.)
> Package opentracer is in "Maintenance" mode and limited support is offered. Please consider using OpenTelemetry or ddtrace/tracer directly.